fold/fold.go

```diff
@@ -0,0 +1,895 @@
+package fold
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/TimothyStiles/poly/transform"
+)
+
+// Fold the DNA sequence and return the lowest free energy score.
+//
+// Based on the approach described in:
+// Zuker and Stiegler, 1981
+// https://www.ncbi.nlm.nih.gov/pmc/articles/PMC326673/pdf/nar00394-0137.pdf
+//
+// If the sequence is 50 or more bp long, "isolated" matching bp
+// are ignored in V(i,j). This is based on an approach described in:
+// Mathews, Sabina, Zuker and Turner, 1999
+// https://www.ncbi.nlm.nih.gov/pubmed/10329189
+// Args:
+//
+//	seq: The sequence to Fold
+//	temp: The temperature the Fold takes place in, in Celsius
+//
+// Returns a slice of NucleicAcidStructure with the energy and description,
+// i.e. stacks, bulges, hairpins, etc.
+func Fold(seq string, temp float64) ([]NucleicAcidStructure, error) {
+	foldContext, err := NewFoldingContext(seq, temp)
+	if err != nil {
+		return nil, fmt.Errorf("error creating folding context: %w", err)
+	}
+
+	// get the minimum free energy structure out of the cache
+	return Traceback(0, len(seq)-1, foldContext), nil
+}
+
+// MinimumFreeEnergy folds the sequence and return just the delta G of the structure
+// Args:
+//
+//	seq: The sequence to fold
+//	temp: The temperature to fold at
+//
+// Returns: the minimum free energy of the folded sequence
+func MinimumFreeEnergy(seq string, temp float64) (float64, error) {
+	NucleicAcidStructures, err := Fold(seq, temp)
+	if err != nil {
+		return 0, fmt.Errorf("error folding: %w", err)
+	}
+	summedEnergy := 0.0
+	for _, structure := range NucleicAcidStructures {
+		summedEnergy += structure.E
+	}
+	return RoundFloat(summedEnergy, 2), nil
```

Comment:
why the round float here?

Reply:
It was there in the python code, I assume it was rounded to be pretty printed.
I think we can safely remove it.